src/main/java/com/elixr/poc/service/PurchaseService.java

```diff
@@ -15,6 +20,25 @@ public PurchaseService(PurchaseRepository purchaseRepository) {
         this.purchaseRepository = purchaseRepository;
     }
 
+    public void createPurchase(Purchase purchase) {
+        savePurchase(purchase);
+    }
+
+    /**
+     * Calling purchase repository to store the valid user information to the database
+     *
+     * @param purchase
+     * @return
+     */
+    private Purchase savePurchase(Purchase purchase) {
+        if (purchase.getId() == null || purchase.getId().toString().isEmpty()) {
+            purchase.setId(UUID.randomUUID());
+        }
+        purchase = this.purchaseRepository.save(purchase);
+        return purchase;
+        this.purchaseRepository = purchaseRepository;
```

Comment:
why this assignment

Reply:
now, removed it, and changed the return type to void.